Building executable with same basename as input source fails

```
% gerbil version
Gerbil v0.18 on Gambit v4.9.5-40-g24201248
```

We try to compile `hello.ss` into `hello`:

```
% cat hello.ss
(export main)
(def (main . args)
  (displayln "hello, world"))
% gxc -exe -o hello hello.ss
/tmp/gxc.1700324049.6964028/hello.scm:
/tmp/empty/hello.scmx:
/tmp/gxc.1700324049.6964028/hello.c:
/tmp/empty/hello.c:
/tmp/empty/hello_.c:
/usr/bin/ld: /tmp/empty/hello.o: in function `___LNK_hello':
hello.c:(.text+0x3d2): multiple definition of `___LNK_hello'; /tmp/gxc.1700324049.6964028/hello.o:hello.c:(.text+0x1da): first defined here
collect2: error: ld returned 1 exit status
*** ERROR IN gxc#compile-executable-module/separate -- 
*** ERROR IN ?
--- Syntax Error at compile: Compilation error; process exit with nonzero status
... form:   "gcc"
```

However, naming the output binary differently makes it work:

```
% gxc -exe -o myhello hello.ss 
/tmp/gxc.1700324120.77565/hello.scm:
/tmp/empty/myhello.scmx:
/tmp/gxc.1700324120.77565/hello.c:
/tmp/empty/myhello.c:
/tmp/empty/myhello_.c:
% ./myhello 
hello, world
```

(Personally, I think the name of the output file should not change the binary content at all.)

--- a/src/gerbil/compiler/driver.ss
+++ b/src/gerbil/compiler/driver.ss
@@ -184,9 +184,8 @@ namespace: gxc
            (output-scm       (path-expand output-scm))
            (output-c         (replace-extension output-scm ".c"))
            (output-o         (replace-extension output-scm ".o"))
-           (output_          (string-append (path-strip-extension output-scm) "_"))
-           (output_-c        (string-append output_ ".c"))
-           (output_-o        (string-append output_ ".o"))
+           (output_-c        (replace-extension output-scm "_.c"))
+           (output_-o        (replace-extension output-scm "_.o"))
            (gsc-link-opts    (gsc-link-options))
            (gsc-cc-opts      (gsc-cc-options))
            (gsc-static-opts  (gsc-static-include-options gerbil-staticdir))
@@ -239,7 +238,7 @@ namespace: gxc
         (delete-directory* tmp))))
 
   (let* ((output-bin (compile-exe-output-file ctx opts))
-         (output-scm (string-append output-bin ".scmx")))
+         (output-scm (string-append output-bin "__exe.scm")))
     (compile-stub output-scm output-bin)
     (unless (current-compile-keep-scm)
       (delete-file output-scm))))
@@ -364,7 +363,7 @@ namespace: gxc
                  default-ld-options ...]))))
 
   (let* ((output-bin (compile-exe-output-file ctx opts))
-         (output-scm (string-append output-bin ".scmx")))
+         (output-scm (string-append output-bin "__exe.scm")))
     (compile-stub output-scm output-bin)
     (unless (current-compile-keep-scm)
       (delete-file output-scm))))
@@ -851,11 +850,7 @@ namespace: gxc
   (cond
    ((pgetq output-file: opts) => values)
    (else
-    (let (mod-str (symbol->string (expander-context-id ctx)))
-      (cond
-       ((string-rindex mod-str #\/)
-        => (lambda (ix) (substring mod-str (fx1+ ix) (string-length mod-str))))
-       (else mod-str))))))
+    (path-strip-directory (symbol->string (expander-context-id ctx))))))
 
 (def (static-module-name idstr)
   (cond

